flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java

```diff
@@ -141,7 +169,12 @@ public void eval(Object... keys) {
 			try {
 				statement.clearParameters();
 				for (int i = 0; i < keys.length; i++) {
-					JDBCUtils.setField(statement, keySqlTypes[i], keys[i], i);
+					if (containsNull) {
+						JDBCUtils.setField(statement, keySqlTypes[i], keys[i], 2 * i);
+						JDBCUtils.setField(statement, keySqlTypes[i], keys[i], 2 * i + 1);
```

Comment:
We can keep this special logic for now. But please open a JIRA to improve this. We can introduced a custom `NamedPreparedStatement` to pass each field only once. see https://www.javaworld.com/article/2077706/named-parameters-for-preparedstatement.html